common/workunit/workunit.cpp

```diff
@@ -167,6 +167,29 @@ void doDescheduleWorkkunit(char const * wuid)
     do more = root->removeProp(xpath.str()); while(more);
 }
 
+StringBuffer & getThorLogDateFromLogFileName(const char * logFileName, const char * masterLogKeyWord, StringBuffer & logDate)
+{
+    //The existing thor master stores the information about a WU into thor master logs
+    //which are named like '...thormaster.yyyy_mm_dd.log'. The datePtr is pointed at
+    //the end of the 'thormaster.'. After that, there is the 'yyyy_mm_dd' part which 
+    //is 10 bytes long. The strlen(datePtr) should >= 10.
+    //If thor master stores all of its log lines for a WU into one log file, there is
+    //no need for the log file name to have the 'yyyy_mm_dd' part. The log name may
+    //look like '...thormaster.log'. The datePtr is pointed at the end of the 'thormaster.'.
+    //The strlen(datePtr) should = 3.
+    //So, if (strlen(datePtr) >= 10), we read the logDate (10 bytes) at the datePtr.
```

Comment:
am I correct in thinking this method is only used for legacy workunits, that don't have the new @pattern ?

Reply:
I think that logic should be rolled into getWUThorLogInfo, i.e. it should deduce by whatever info. available regardless of whether imported or non-imported, and should make the log date available from IConstWUThorLogInfo.

Reply:
After my new changes, this function (getThorLogDateFromLogFileName()) will be called by 2 palces: 1. getWUThorLogInfo(); 2. getSlaveLogPatternForLegacyWU() when importing a ZAP.